### ISSUE TITLE
Screen Reader stops reading content after 10000 characters

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#96153]
Issue Type: <b>Bug</b>

This bug was shown to me by @jeanbrazcosta using NVDA, but it also happens with orca.

1. Launch orca or Nvda.
2. Open the attached file using VSCode.
3. Jump to line 23 pressing ctrl+g.
4. Press the down arrow a few times.

Although there is content on the lines that follow line 23, the screen reader reads nothing.
It seems to me that the problem has something to do with the size of line 23, but I cannot say. The file has 25 lines and the length of line 23 is approximately 7485 characters.

CC @isidorn 
VS Code version: Code - Insiders 1.45.0-insider (a250df703de955a38aed427a917bce8278ab3331, 2020-04-24T09:57:20.729Z)
OS version: Linux x64 5.6.6-arch1-1

[tx.txt](https://github.com/microsoft/vscode/files/4534222/tx.txt)

**Comment by isidorn** [comment]
@jvesouza thanks for filling this. This is an upstream issue in Chrome
https://bugs.chromium.org/p/chromium/issues/detail?id=1001591&q=10000%20chars&can=2

As a workaround I suggest that you put `"editor.accessibilityPageSize": 20` in your settings. Or any other lower number.

I belive colleagues of @joanmarie are working on a related fix on Chrome that as a result might fix the upstream issue. In case that happens we plan to backport the Chrome fix to our Electron Chrome.

**Comment by jvesouza** [comment]
@isidorn  As soon as I put the suggested parameter in settings.json the problem disappeared as if by magic! can there be any collateral effect for this workaround?

@jeanbrazcosta Could you check?

**Comment by isidorn** [comment]
@jvesouza the collateral is that we now only set 20 lines to our hidden accessibility textArea, thus if you tell your screen reader to read the content of the whole editor it will read only 20 lines.

Note that this 20 lines get updated as you navigate in the document, so all should work nicely except the "say all" command.

**Comment by jvesouza** [comment]
@isidorn Thanks for the clarification, I always wanted to ask what this parameter was for. 
Since I rarely use the say all command to read the entire code, I don't think it will be a big problem for me.

**Comment by jvesouza** [comment]
@isidorn In the defaultSettings.json I found the following:
"editor.accessibilityPageSize": 10,

This is OK? Shouldn't it be 100?

Is there any performance gain or loss if that number is lower?
Thanks.

**Comment by isidorn** [comment]
By default it is 10, we set it to 100 when we detect a screen reader since we thought that would make a nicer screen reader experience due to the "say all" command.
However due to the issue you are hitting we might have been wrong.

There is potentialy a perf gain for lower numbers, thus we put 10 for all non screen reader users.
Imho the perf change will not be noticable.

**Comment by joanmarie** [comment]
@isidorn: Regarding this:

> I belive colleagues of @joanmarie are working on a related fix on Chrome that as a result might fix the upstream issue. In case that happens we plan to backport the Chrome fix to our Electron Chrome.

I think this may already be fixed (by someone else). I'll comment on the bug.

**Comment by jeanbrazcosta** [comment]
> @jvesouza the collateral is that we now only set 20 lines to our hidden accessibility textArea, thus if you tell your screen reader to read the content of the whole editor it will read only 20 lines.
> 
> Note that this 20 lines get updated as you navigate in the document, so all should work nicely except the "say all" command.

@jvesouza and @isidorn, I did a test here by putting all the required settings but for me it didn't work
I'm using NVDA.
This is the file that I am trying to edit, and I can't seem to read that line 54
[index.txt](https://github.com/microsoft/vscode/files/4542722/index.txt)

**Comment by jvesouza** [comment]
@isidorn  I tested with the file attached by @jeanbrazcosta and had no problems. However, I found that if I change the parameter value from 20 to 10 or 50, the problem appears again.
It seems to me that there is no simple way to correct this because the issue is not the number of lines but the number of characters. I understand that if the total number of characters in the 20 lines to be placed in the buffer exceeds 10,000, the problem will arise.
Would it be possible to include some logic to only include lines in the buffer if the limit of 10000 was not exceeded?

**Comment by isidorn** [comment]
@jvesouza yes, we could do some workaround of a 10000 limit, however the proper solution would be to just back port the fix from Chrome to our electron version. So I will first see if that is possible.

**Comment by jeanbrazcosta** [comment]
I would like to point out that, I tested with several parameters ie
pagesize 20, 10 and for me the problem has not disappeared.

Em ter., 28 de abr. de 2020 às 04:43, Isidor Nikolic <
notifications@github.com> escreveu:

**Comment by jvesouza** [comment]
@jeanbrazcosta 
* How did you specify this parameter? Using the VSCode interface or directly editing the settings.json file?
* What version of VSCode are you using?
* Would be possible to test using the insiders version?
* What are the steps to reproduce the problem on your machine?

**Comment by jeanbrazcosta** [comment]
I'm sorry, I changed the version to "insiders", and it really worked for
you when you specify all the parameters

Em ter., 28 de abr. de 2020 às 08:58, José Vilmar Estácio de Souza <
notifications@github.com> escreveu:

**Comment by isidorn** [comment]
I believe this should no longer happen with latest VS Code. Chrome has picked up a fix for this.
I can no longer reproduce this,  thus optimistically closing thus. If somebody still see this please let me know.

I have also changed the accessibilityPageSize to be 2000 by default when a screen reader is detected

**Comment by rebornix** [comment]
I tested with narrator:

* With 1.45, when I move the focus to line 23, it said blank and VS Code froze
* With latest Insiders, it works as same as other lines, arrow down worked too.
